From 805181c589f2f5bb71a9245b3f9dd00300e8026c Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Fri, 17 Oct 2025 22:16:38 +0200
Subject: [PATCH 1/3] Allow to use system robin-map instead of bundled

Adds new HEAPTRACK_USE_SYSTEM_ROBINMAP CMake option

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
(cherry picked from commit a7bedb1b28fef4e7976269e9f47c0f62d90b4836)
---
 3rdparty/CMakeLists.txt |  7 ++++---
 CMakeLists.txt          | 10 ++++++++++
 2 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/3rdparty/CMakeLists.txt b/3rdparty/CMakeLists.txt
index a2c8e69..297e227 100644
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -18,6 +18,7 @@ if (ZSTD_FOUND AND NOT BOOST_IOSTREAMS_HAS_ZSTD)
     mark_as_system_target(boost-zstd)
 endif()
 
-add_subdirectory(robin-map)
-
-mark_as_system_target(robin_map)
+if(NOT HEAPTRACK_USE_SYSTEM_ROBINMAP)
+    add_subdirectory(robin-map)
+    mark_as_system_target(robin_map)
+endif()
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6893e81..a3dedd0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,6 +111,12 @@ option(
   ON
 )
 
+option(
+  HEAPTRACK_USE_SYSTEM_ROBINMAP
+  "Use system robin-map instead of bundled"
+  OFF
+)
+
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 if (NOT MSVC)
@@ -170,6 +176,10 @@ if (HEAPTRACK_BUILD_TRACK)
     endif()
 endif()
 
+if (HEAPTRACK_USE_SYSTEM_ROBINMAP)
+    find_package(tsl-robin-map CONFIG REQUIRED)
+endif()
+
 add_subdirectory(3rdparty)
 add_subdirectory(src)
 add_subdirectory(tests)
-- 
2.55.0

